go-controller/pkg/ovn/obj_retry.go

```diff
@@ -917,15 +917,18 @@ func (oc *Controller) iterateRetryResources(r *retryObjs, updateAll bool) {
 // periodicallyRetryResources tracks retryObjs and checks if any object needs to be retried for add or delete every
 // retryObjInterval seconds or when requested through retryChan.
 func (oc *Controller) periodicallyRetryResources(r *retryObjs) {
+	timer := time.NewTicker(retryObjInterval)
+	defer timer.Stop()
 	for {
 		select {
-		case <-time.After(retryObjInterval):
+		case <-timer.C:
 			klog.V(5).Infof("%s s have elapsed, retrying failed objects of type %v", retryObjInterval, r.oType)
 			oc.iterateRetryResources(r, false)
 
 		case <-r.retryChan:
 			klog.V(5).Infof("Retry channel got triggered: retrying failed objects of type %v", r.oType)
 			oc.iterateRetryResources(r, true)
+			timer.Reset(retryObjInterval)
```

Comment:
as I understand the new API, we:
- create a new ticker with a timeout
- wait on its channel 
- once the channel gets triggered, we do our stuff (in this case `iterateRetryResources`) and then we reset the timer

So I would put `timer.Reset(retryObjInterval)` at the end of the previous case block where the channel is consumed (`case <-timer.C:`) and not here, unless I am missing something :)

Reply:
No, the timer resets by itself when it expires. But when a different event trigger, different than the timer, that makes us do the same thing we would do when the timer expires, we need to reset the timer to not do that thing again in less than the timer interval.

Reply:
All right, it makes sense then.